kerberos.go

```diff
@@ -0,0 +1,137 @@
+package hdfs
+
+import (
+	"log"
+	"os"
+	"os/user"
+	"strings"
+
+	"gopkg.in/jcmturner/gokrb5.v3/client"
+	"gopkg.in/jcmturner/gokrb5.v3/config"
+	"gopkg.in/jcmturner/gokrb5.v3/credentials"
+	"gopkg.in/jcmturner/gokrb5.v3/keytab"
+)
+
+const HdfsDefaultServiceName = "nn"
+const KrbDefaultCfgPath = "/etc/krb5.conf"
+const HadoopAuthCfgPath = "hadoop.security.authentication"
+const HdfsDefaultConfDir = "/etc/hadoop/conf"
+
+// GetKrbClientIfRequired returns a kerberos client if the hadoop
+// configuration or the environment variables suggest one is required
+func GetKrbClientIfRequired(conf HadoopConf) *client.Client {
```

Comment:
I think better would be to create a generalized `SetConnectionOptions` method on `HadoopConf`. Something like

```
// SetConnectionOptions sets the necessary namenode address(es) and authentication settings on clientOptions from the loaded configuration, or returns an error if not enough information is present in the configuration to do so.
func (conf *HadoopConf) SetConnectionOptions(clientOptions ClientOptions) error { ... }
```

Reply:
Agreed. Though, would it not make more sense to call this method on the ClientOptions structe? ie, have:

```
func (opts *ClientOptions) SetConnectionOptions(conf *HadoopConf)
```

(I'm a bit bothered by having a `Set` function on something that sets something in the parameter, but if that's a typical way to do it in Go why not :))

Reply:
You're right, that's better.